crates/core/tedge_actors/src/builders.rs

```diff
@@ -213,82 +197,6 @@ pub trait RuntimeRequestSink {
     fn get_signal_sender(&self) -> DynSender<RuntimeRequest>;
 }
 
-/// A trait that defines that an actor provides a service
-/// by accepting `Request` messages and sending `Response` from/to its peers.
-///
-/// In order to connect to a to `ServiceProvider<Req, Res, Conf>` and avail its services,
-/// the peer must be a `ServiceConsumer<Req, Res, Conf>`.
-///
-/// The config parameter is typically used by the `ServiceConsumer`
-/// to register any message filtering criteria to the `ServiceProvider`.
-pub trait ServiceProvider<Request: Message, Response: Message, Config> {
```

Comment:
Deprecating this trait and its companion, the `ServiceConsumer` trait, is the heart of this PR.

There are two cases to consider when updating an actor which was using this trait.

### First case

When there is no request/response relationship between the message received and those sent by the provider actor. This is for instance the case for the MQTT messages received from and published to an MQTT actor.  An actor builder that needs to connect such an MQTT actor, will no more require a service provider:

~`mqtt: &mut impl ServiceProvider<MqttMessage, MqttMessage, TopicFilter>`~

but connect an actor that is a source of MQTT messages (received from the broker) as well as a sink of MQTT messages (to be published to the broker).


```
mqtt: &mut (impl MessageSource<MqttMessage, TopicFilter> + MessageSink<MqttMessage, NoConfig>),
```

### Second case

When there is a request/response relationship between the message received and those sent by the provider actor. This is for instance the case for the SetTimeout requests to the timer actor and its Timeout responses.

The expectation on the actor to connect to is no more a

~`timer: &mut impl ServiceProvider<SyncStart, SyncComplete, NoConfig>`~

but:

```
timer: &mut impl Service<SyncStart, SyncComplete>,
```